### DISCUSSION TITLE
option to refetch only first group of useInfinteQuery results?

### DISCUSSION THREAD

**Discussion by skulegirl** [TanStack/query#858] [General]
With useInifiniteQuery, currently if I have already loaded N pages, then calling refetch will result in N API calls to refetch all the data. I understand that this is by design.

For a react native app, I want a pull-from-top refresh to continue to display only the first page while it is refreshed, but remove all subsequent pages from the cache and not refetch them unless the fetchMore method is called again. Can we add a way to reset the number of loaded groups in the useInfiniteQuery cache data? I've tried using queryCache.setQueryData to delete all but the first group of data, but it's not sufficient as it still seems to know that more groups were previously fetched and then proceeds to refetch them.

**Comment by dpyzo0o** [comment]
I have the exact use case, would like to see some solid solutions.

**Comment by skulegirl** [comment]
If there's no known way to accomplish this then I can turn this into an issue rather than a discussion for hopes of a feature request.

**Reply by dpyzo0o** [reply]
I think it makes sense. Please do so,  thanks

**Reply by JeremyBradshaw7** [reply]
Did you do that @skulegirl ? I have the same use-case, and cannot find an issue raised.

**Reply by skulegirl** [reply]
Somehow it ended up as another discussion instead of an issue (https://github.com/tannerlinsley/react-query/discussions/875). But I can confirm that the changes for #914 does seem to fix it for me.
